test/support/functional_assertion_helpers.ex

```diff
@@ -0,0 +1,57 @@
+defmodule SSHKit.FunctionalAssertionHelpers do
+  @moduledoc false
+  import ExUnit.Assertions
+  alias SSHKit.Context
+  alias SSHKit.SSH
+
+  def verify_transfer(conn, local, remote) do
+    command = &"find #{&1} -type f -exec #{&2} {} \\; | sort | awk '{print $1}' | xargs"
+    compare_command_output(conn,
+      command.(local, SystemCommands.shasum_cmd()),
+      command.(remote, "sha1sum")
+      )
+  end
+
+  def verify_mode(conn, local, remote) do
+    command = &"find #{&1} -type f -exec ls -l {} + | awk '{print $1 $5}' | sort |xargs"
+    compare_command_output(conn,
+      command.(local),
+      command.(remote)
+      )
+  end
+
+  def verify_atime(conn, local, remote) do
+    command = &"env find #{&1} -type f -exec #{&2} {} \\; | cut -f1,2 | sort | xargs"
+      compare_command_output(conn,
+        command.(local, SystemCommands.stat_cmd()),
+        command.(remote, "stat -c '%s\t%X\t%Y'")
+      )
+  end
+
+  def verify_mtime(conn, local, remote) do
+    command = &"env find #{&1} -type f -exec #{&2} {} \\; | cut -f1,3 | sort | xargs"
+    compare_command_output(conn,
+      command.(local, SystemCommands.stat_cmd()),
+      command.(remote, "stat -c '%s\t%X\t%Y'")
+      )
+  end
+
+  def compare_command_output(context = %Context{}, local, remote) do
+    Enum.map(context.hosts,
+      fn(h) ->
+        {:ok, conn} = SSH.connect(h.name, h.options)
```

Comment:
Could we also make use of the new "lambda" *(still don't know if that's semantically correct...)* syntax here?

Something like this:
```elixir
SSH.connect h.name, h.options, fn(conn) ->
  compare_command_output(conn, local, remote)
end
```
This would spare the extra `SSH.close`, but I'm not sure if it makes the helper less complex after all... What do you think?